Trakt/Helpers/UserDataManagerEventsHelper.cs

```diff
@@ -159,12 +159,28 @@ public void ProcessUserDataSaveEventArgs(UserDataSaveEventArgs userDataSaveEvent
             }
         }
 
-        _userDataPackages[traktUser.LinkedMbUserId] = userPackage;
+        lock (_userDataPackages)
+        {
+            _userDataPackages[traktUser.LinkedMbUserId] = userPackage;
+        }
     }
 
     private void OnTimerCallback(object state)
     {
-        foreach (var package in _userDataPackages)
+        Dictionary<string, UserDataPackage> userDataQueue;
+        lock (_userDataPackages)
+        {
+            if (!_userDataPackages.Any())
+            {
+                _logger.LogInformation("No events... stopping queue timer");
+                return;
+            }
+
+            userDataQueue = _userDataPackages;
+            _userDataPackages.Clear();
```

Comment:
Does this work? You're just saving a reference to a local variable and then clearing the dictionary. `userDataQueue` will be empty then?

Reply:
You're correct, I created a new dict from the current one that seems to work.